src/algorithm/verifySolidAccessToken.ts

```diff
@@ -70,7 +65,7 @@ export async function verifySolidAccessToken(
    *    - 'iat' is not in the future
    */
   const { payload, protectedHeader } = await jwtVerify(
-    accessTokenValue,
+    `${solidJwt.joseHeader}.${solidJwt.jwsPayload}.${solidJwt.jwsSignature}`,
```

Comment:
It's weird that we deconstruct and then reconstruct. Now the knowledge about header structure resides in two places. Should `parseSolidAuthorizationHeader` just return that then?

Reply:
I thought about returning a "value" field in addition to the parts, but I wasn't  not convinced it's nicer. Would you prefer that?

I wouldn't just return the value, because we use the segments separately for validation.

Reply:
> I thought about returning a "value" field in addition to the parts, but I wasn't not convinced it's nicer. Would you prefer that?

I'd do that, yeah; the the consuming class needs zero knowledge of what the token looks like.

Reply:
You're right. Implemented: https://github.com/solid/access-token-verifier/pull/20/commits/8c75ee54e096d9cc83220cd02c279b8c920f3000